### PR TITLE
Add isInstanceOf for generic procs to the macros module

### PR DESCRIPTION
**Pull request by cooldome**
In typed macros we often compare symbols. Comparison often made with the `bindSym` output. However this comparison does not work if a symbol belongs to the generic proc. In the proc bodies we are analysing, we encounter only already resolved instances of generic procs while `bindSym` returns generic proc symbols. These are understandably are different symbols and comparison returns `false`. AFAIK, there is no way at the moment check if proc symbol is the instance of a particular generic proc. This PR adds this functionality.

### CODE REVIEW

**Comment by cooldome** [comment]
Test failures are unrelated

**Comment by krux02** [comment]
I don't think this is a good name for it. `isInstanceOf` is already a name in the scala programming language, but here it is to check if an object is from a given type. People who come from scala (like me) would assume something very different when they see thing function. I would suggest a name like: ``isGenericProcInstanceOf``

**Comment by cooldome** [comment]
I have renamed the new proc to ``isGenericProcInstanceOf`` as you suggested

**Comment by Araq** [comment]
Er, sorry but the name should be ``isInstantiationOf`` or ``isGenericProcInstantiationOf`` or similar, "instantiation", not "instance", but maybe it's nitpicking.

**Comment by cooldome** [comment]
Ok, I will rename to isInstantiationOf

**Comment by cooldome** [comment]
Test failures are unrelated, please review

File: changelog.md [hunk 1]
```diff
@@ -61,6 +61,10 @@ proc enumToString*(enums: openArray[enum]): string =
 - Added `system.typeof` for more control over how `type` expressions
   can be deduced.
 
+- Added `macros.isGenericProcInstanceOf` for checking if the proc symbol 
+  is instance of generic proc symbol.
```
Comment (narimiran) [inline changelog.md]: instance --> instantiation (2x)

File: compiler/vm.nim [hunk 2]
```diff
@@ -959,7 +959,18 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
                         else: newSymNode(a.sym.skipGenericOwner)
         regs[ra].node.flags.incl nfIsRef
       else:
-        stackTrace(c, tos, pc, "node is not a symbol")
+        stackTrace(c, tos, pc, "node is not a symbol")#
```
Comment (narimiran) [inline compiler/vm.nim]: delete `#`

Comment (cooldome) [reply compiler/vm.nim]: fixed

File: changelog.md (hunk 1 above)
Comment (cooldome) [reply changelog.md]: fixed

**Comment by cooldome** [comment]
All comments resolved, ready for committer

**Comment by cooldome** [comment]
test failures are unrelated

**Pull request merged**